docker-git-sync.py

```diff
@@ -1,133 +1,169 @@
-                            os.remove(link)
-                            logging.info(f"Deleted file/link: {link}")
-                        os.symlink(source, link)
-                        logging.info(f"Created link from {link} -> {source}.")
-
-
-def execute_post_commands(post_commands):
+        compose_file = stack.get("compose") or "docker-compose.yaml"
+        if not workdir:
+            continue
+        env = stack.get("values")
+        program = command.split(" ") + ["-f", compose_file, "up", "-d"]
+        logging.info(f"Executing {program} with env: {env}")
+        try:
+            result = check_output(program, cwd=workdir, env=env).decode("utf-8").strip()
+        except (RuntimeError, CalledProcessError) as error:
+            logging.error(f"Runtime Error occurred while executing:\n {error}")
+            return
+        if result != "":
+            logging.info(result)
+
+
+def link_nginx_configs(content: dict[str, Any]) -> None:
+    if not content.get("nginx"):
+        return
+    if not content["nginx"].get("config"):
+        return
+
+    nginx_config = content["nginx"]["config"]
+    source = nginx_config.get("provisioning")
+    target = nginx_config.get("target")
+    sites_enabled = nginx_config.get("sites-enabled")
+
+    if not sites_enabled.endswith("/"):
+        sites_enabled += "/"
+    if target and source:
+        # Backup old file because in next step it would be overwritten.
+        if Path(target).is_file():
+            shutil.move(target, target + ".bak")
+            logging.info(f"Moved {target} to {target}.bak (backup file).")
+
+        # Overwrite nginx config file.
+        shutil.copy(source, target)
+        logging.info(f"Copied {source} to {target}.")
+
+    nginx_sites = content["nginx"].get("sites")
+    if not nginx_sites:
+        return
+    for site_name in nginx_sites:
+        site = nginx_sites.get(site_name)
+        link = Path(sites_enabled) / site.get("name")
+        source = Path(site.get("file")).resolve()
+        if link.resolve() == source:
+            continue
+        if link.is_symlink() or link.is_file():
+            link.unlink()
+            logging.info(f"Deleted file/link: {link}")
+        link.symlink_to(source)
+        logging.info(f"Created link from {link} -> {source}.")
+
+
+def execute_post_commands(post_commands: list[str]) -> None:
     for command in post_commands:
         logging.info(f"Executing {command}")
         result = check_output(command.split(" ")).decode("utf-8").strip()
-        if result != "":
+        if result:
             logging.info(result)
 
 
-def find_yaml_config(args):
+def find_yaml_config(args) -> None:
     # For loop to search for yaml file in servers.
-    for file in walk_through_files(path=args.server_directory):
-        content = yaml.safe_load(open(file))
+    server_directory = Path(args.server_directory)
+    for file in server_directory.rglob("*.yaml") + server_directory.rglob("*.yml"):
```

Comment:
This does not work for me: 
```
Traceback (most recent call last):
  File "/mnt/f/workspace/docker-compose-ops/docker-git-sync.py", line 171, in <module>
    main()
  File "/mnt/f/workspace/docker-compose-ops/docker-git-sync.py", line 167, in main
    find_yaml_config(args)
  File "/mnt/f/workspace/docker-compose-ops/docker-git-sync.py", line 102, in find_yaml_config
    for file in server_directory.rglob("*.yaml") + server_directory.rglob("*.yml"):
TypeError: unsupported operand type(s) for +: 'generator' and 'generator'
```

Possible fix:
```suggestion
    from itertools import chain
    
    for file in chain(server_directory.rglob("*.yaml"), server_directory.rglob("*.yml")):
```

Reply:
TIL. Good catch 🐈‍⬛